docs/playground/ko/TypeScript/Meta-Types/Mapped Types.ts

```diff
@@ -0,0 +1,57 @@
+// 매핑된 타입은 다른 타입 기반의 새로운 타입을 만드는 방법입니다.
+// 또한 효과적인 변형 타입입니다.
+
+// 매핑된 타입을 사용하는 일반적인 케이스는
+// 기존 타입의 부분적인 하위집합을 다루는 것입니다.
+// 예를 들어 API는 Artist를 반환할 수 있습니다:
+
+interface Artist {
+  id: number;
+  name: string;
+  bio: string;
+}
+
+// 그러나, Artist의 하위집합만 변경하는 API에
+// 수정사항을 보냈었다면
+// 일반적으로 추가 타입을 만들어야만 했습니다:
+
+interface ArtistForEdit {
+  id: number;
+  name?: string;
+  bio?: string;
+}
+
+// 위에 Artist와 같지 않을 수도 있습니다.
+// 매핑된 타입으로 기존 타입에서
+// 변화를 만들어 낼 수 있습니다.
+
+type MyPartialType<Type> = {
+  // 모든 기존 프로퍼티에 대해서
+  // Type의 타입 내부는 ?: 버전으로 변환합니다
+  [Property in keyof Type]?: Type[Property];
+};
+
+// 이제 edit 인터페이스를 만드는 대신에
+// 매핑된 타입을 사용할 수 있습니다:
+type MappedArtistForEdit = MyPartialType<Artist>;
+
+// 완벽에 가깝지만,
+// id가 절대 발생하면 안되는 null이 될 수 있게 해줍니다.
```

Comment:
```suggestion
// id 값이 null 이 되는 상황을 방지할 수 없습니다.
```
현재 문맥에서 의미하는 바를 강조하기 위해 의역으로 제안합니다.